### PR TITLE
add sst file iterator

### PR DESCRIPTION
**Pull request by bobotu**
refer to #107

### CODE REVIEW

File: rocksdb/sst_file_iterator.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+	indexBlockIter *blockIterator
+	dataBlockIter  *blockIterator
+	dataBuf        []byte
+	invalid        bool
+	err            error
+	checksumType   ChecksumType
+}
+
+func NewSstFileIterator(f *os.File) *SstFileIterator {
+	return &SstFileIterator{
```
Comment (coocood) [inline rocksdb/sst_file_iterator.go]: I think it's better to load the footer in this function.
`SeekToFirst` may be called multiple times.

File: rocksdb/sst_file_iterator_test.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		key := it.Key()
+		value := string(it.Value())
+
+		require.Equal(t, nums[i], string(key.UserKey))
+		require.Equal(t, nums[i], string(value))
+		i++
+	}
+	require.Equal(t, num, i)
+	require.Nil(t, it.Err())
+}
```
Comment (coocood) [inline rocksdb/sst_file_iterator_test.go]: We need to test that `SeekToFirst` called more than once.

File: rocksdb/sst_file_iterator.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+	if it.dataBuf, err = it.decompressBlock(it.dataBuf); err != nil {
+		return err
+	}
+	it.dataBlockIter.Reset(it.dataBuf)
+
+	return nil
+}
+
+func (it *SstFileIterator) changeBufSize(sz uint64) {
+	if uint64(len(it.dataBuf)) < sz {
```
Comment (coocood) [inline rocksdb/sst_file_iterator.go]: Use `cap` instead of `len` can save some memory allocation.

File: rocksdb/sst_file_iterator.go [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+	trailerPos := len(data) - blockTrailerSize
+
+	blkData := data[:trailerPos]
+	compressTp := CompressionType(data[trailerPos])
+
+	switch it.checksumType {
+	case ChecksumCRC32:
+		crc := newCrc32()
+		crc.Write(data[:trailerPos])
+		crc.Write([]byte{data[trailerPos]})
```
Comment (coocood) [inline rocksdb/sst_file_iterator.go]: Why not ```crc.Write(data[:trailerPos+1]```?

File: rocksdb/sst_file_iterator.go (hunk 3 above)
Comment (bobotu) [reply rocksdb/sst_file_iterator.go]: In this case `len(buf)` always equals to `cap(buf)`.

File: rocksdb/sst_file_iterator.go (hunk 4 above)
Comment (bobotu) [reply rocksdb/sst_file_iterator.go]: Did you mean `crc.Write(data[trailerPos:trailerPos+1])`?

Comment (coocood) [reply rocksdb/sst_file_iterator.go]: Then we can call `Write` only once.

**Comment by coocood** [comment]
LGTM

**Pull request merged**